### DISCUSSION TITLE
Koishi 4.11.0

### DISCUSSION THREAD

**Discussion by shigma** [koishijs/koishi#898] [General]
## Notable Changes

- 此版本中我们移除了 @koishijs/cli 包，将其合并到了 koishi 中。这意味着你每次升级时不再需要同时升级两边了 (koishijs/koishi@0520ba8dc902227675aea5f66c78666d8687708d, koishijs/koishi@48feca2bdc64cb20d217b5dd083602442ae5c16e)
  - 请注意：这个改动可能会导致你本地的启动脚本失效。为此我们发布了一个专门的插件 koishi-plugin-migration 来尝试此类问题。**要升级 Koishi 的用户可以尝试先安装此插件，然后在插件的面板中点击「一键升级」**。目前此插件还在测试阶段，未来可能将作为 @koishijs/plugin-market 的内置功能。
- 废弃了 `base64:` 协议，转为更标准和通用的 `data:` 协议 (satorijs/satori@106b821da37309d95379f4e9606670d74fa9889b)
  - 对于 `h.image(buffer)`，现在需要你额外传入一个 mime type 作为第二个参数，例如 `h.image(buffer, 'image/png')`
  - 过去的写法目前依然可用，但相关的支持会在未来的版本中被移除

## Features

- **core:** 支持了 `db.drop(table)` 的用法 (shigma/minato#11) (shigma/minato@872a9cf6b10b47491b4b7e2bfef8a0f7cf0c8560)
- **core:** 新增了 `params` 属性，可用于帮助的本地化插值 (koishijs/koishi#845) (koishijs/koishi@5d5f2854695b2de2d38feddd19baa93d5143da35)
- **loader:** 现在 `$if` 后可以直接是表达式，不需要插值了 (koishijs/koishi@637cfdc652100d434a247a3089adb4649b97a48f)
- **market:** 现在能显示 `peerDependencies` 是否可选了 (koishijs/console@521ef2a285d3877104690b9f2896c2e22e3a253f)
- **market:** 优先显示作为 `peerDependencies` 的插件而不是它们对应的服务 (koishijs/console@3be1b0bdea0a431244bcc8013a6eac5eb07e9428)

## Bug Fixes

- **core:** 修复了带选项的 `cmd.shortcut()` 行为错误的问题 (koishijs/koishi#818) (koishijs/koishi@c85f20a7b498394d55be886aa0b9c9c69d5ac616)
- **element:** 修复了 `0` 和 `false` 在插值时会丢失的问题 (satorijs/satori@ce8081ff73dddd58aa45be4153b4605d41d29924)
- **mock:** 修复了 `session.send()` 实现可能报错的问题 (koishijs/koishi@e07ebbcf5b4d5eada756d5c09b0487def05a9574)
- **mongo:** 修复了 `unique` 约束没有忽略空值的问题 (koishijs/koishi#893) (shigma/minato@17c98ea394c610824d59d31000aed9af269d2836)
- **sql:** 修复了启动时可能出现的初始化失败问题 (shigma/minato@268a8e22ae7c69bd88d8162045ae8058aade6c7d)
- **sqlite:** 修复了插入 `Date` 数据可能导致唯一性错误的问题 (koishijs/koishi#667) (shigma/minato#13) (shigma/minato@8ebeff3bd20f99443b53b5b46b31c75d522d3724)

## Other Changes

- chore: bump versions (koishijs/koishi@a8bd6a77a35da390415bcb3143830d3d8751c213)
- chore: fix silly bugs (koishijs/koishi@babe254130a72d9d7584caec3ad250f4fbc312db)
- chore: bump versions (shigma/minato@e70617b5603c939c078ef732cb563431fb050ee4)
- chore: update readme.md (shigma/minato#9) (shigma/minato@15e127aff805c8e9cd60497184841a4bb2666b65)
- chore: bump versions (satorijs/satori@4cf4fcd79285f1cb3ea3af41914efe96393c19ea)
- chore: bump versions (shigma/minato@db08e2c9c97a2575ed37b1144834cb57e8310655)
- chore: bump versions (koishijs/koishi@6115998e054af45520d0cf7262455b1ae01f43a3)
- chore: bump versions (koishijs/console@07da7e8a0a72f380eb8c27f1e9290afc1c96f5ca)

<hr /><em>This discussion was created from the release <a href='https://github.com/koishijs/koishi/releases/tag/4.11.0'>Koishi 4.11.0</a>.</em>

**Comment by neveler** [comment]


**Reply by ilharp** [reply]
重新启动计算机是否可以解决此问题？

**Reply by ilharp** [reply]
这是正常的情况。可以参考本页最上方的 `Notable Changes` 部分进行升级。
